source/common/http/conn_manager_impl.cc

```diff
@@ -831,6 +831,11 @@ void ConnectionManagerImpl::ActiveStream::decodeData(ActiveStreamDecoderFilter*
   }
 
   for (; entry != decoder_filters_.end(); entry++) {
+    // If end_stream_ is marked for a filter, the data is not for this filter and filters after.
+    // Please refer to the `issue <https://github.com/envoyproxy/envoy/issues/5311>`_ for details.
```

Comment:
In this case I would prefer instead of referring to an issue to just spell it out here. Can you briefly summarize why we need to do this?

Reply:
While updating the comment, realized it is better to do such checking in the doData() call. It is more specific.